specs/electra/beacon-chain.md

```diff
@@ -1218,6 +1291,31 @@ def process_attestation(state: BeaconState, attestation: Attestation) -> None:
 
 ##### Deposits
 
+###### Modified `get_validator_from_deposit`
+
+*Note*: The function is modified to use `MAX_EFFECTIVE_BALANCE_ELECTRA` for compounding withdrawal credential.
+
+```python
+def get_validator_from_deposit(pubkey: BLSPubkey, withdrawal_credentials: Bytes32, amount: uint64) -> Validator:
+    if is_compounding_withdrawal_credential(withdrawal_credentials):
+        max_effective_balance = MAX_EFFECTIVE_BALANCE_ELECTRA
+    else:
+        max_effective_balance = MIN_ACTIVATION_BALANCE
+
+    # [Modified in Electra:EIP7251]
+    effective_balance = min(amount - amount % EFFECTIVE_BALANCE_INCREMENT, max_effective_balance)
+
+    return Validator(
+        pubkey=pubkey,
+        withdrawal_credentials=withdrawal_credentials,
+        activation_eligibility_epoch=FAR_FUTURE_EPOCH,
+        activation_epoch=FAR_FUTURE_EPOCH,
+        exit_epoch=FAR_FUTURE_EPOCH,
+        withdrawable_epoch=FAR_FUTURE_EPOCH,
+        effective_balance=effective_balance,
+    )
+```
+
 ###### Modified `apply_deposit`
 
 *Note*: The function `apply_deposit` is modified to support EIP7251.
```

Comment:
`apply_deposit` doesn't seem to be used anymore in Electra. Am I missing anything?

Reply:
It is still used in `process_deposit`. I don't think we are getting rid of yet?

Reply:
Gotcha. Then we should add `process_deposit` here in this spec as *modified*

Reply:
Done!